src/crimson/os/seastore/seastore_types.h

```diff
@@ -642,10 +642,10 @@ struct seg_paddr_t : public paddr_t {
    */
   paddr_t maybe_relative_to(paddr_t base) const {
     assert(!base.is_block_relative());
-    seg_paddr_t& s = base.as_seg_paddr();
-    if (is_block_relative())
+    if (is_block_relative()) {
```

Comment:
This is actually wrong.  Addresses within an extent which refer to other extents within the same record will be block relative whether the extents are in a CBJournal or a SegmentedJournal.  Fix/add a paddr_t::add_block_relative method to behave correctly whether base is random or segmented.